src/docdb/tree/DocDBDatabaseTreeItemBase.ts

```diff
@@ -64,58 +65,53 @@ export abstract class DocDBDatabaseTreeItemBase extends DocDBTreeItemBase<Collec
-                    value: minThroughput.toString(),
-                    ignoreFocusOut: true,
-                    prompt: `Initial throughput capacity, between ${minThroughput} and ${maxThroughput}`,
-                    validateInput: DocDBDatabaseTreeItemBase.validateThroughput
-                }));
-
-                if (throughput) {
-                    const options = { offerThroughput: throughput };
-                    const collectionDef = {
-                        id: collectionName,
-                        partitionKey: {
-                            paths: [partitionKey],
-                            kind: DocumentBase.PartitionKind.Hash
-                        }
-                    };
-
-                    showCreatingNode(collectionName);
-                    const client = this.getDocumentClient();
-                    const collection: CollectionMeta = await new Promise<CollectionMeta>((resolve, reject) => {
-                        client.createCollection(this.link, collectionDef, options, (err, result) => {
-                            err ? reject(err) : resolve(result);
-                        });
-                    });
-
-                    return this.initChild(collection);
-                }
-            }
+        let partitionKey: string | undefined = await ext.ui.showInputBox({
+            prompt: 'Enter the partition key for the collection, or leave blank in case of fixed size.',
+            ignoreFocusOut: true,
+            validateInput: DocDBDatabaseTreeItemBase.validatePartitionKey
+        });
+
+        if (partitionKey && partitionKey.length && partitionKey[0] !== '/') {
+            partitionKey = '/' + partitionKey;
         }
+        if (partitionKey !== "") {
+            collectionDef.partitionKey = {
+                paths: [partitionKey],
+                kind: DocumentBase.PartitionKind.Hash
+            };
+        }
+        const throughput: number = Number(await ext.ui.showInputBox({
+            value: minThroughput.toString(),
+            ignoreFocusOut: true,
+            prompt: `Initial throughput capacity, between ${minThroughput} and ${maxThroughput}`,
+            validateInput: DocDBDatabaseTreeItemBase.validateThroughput
+        }));
+
+        const options = { offerThroughput: throughput };
+
+        showCreatingNode(collectionName);
+        const client = this.getDocumentClient();
+        const collection: CollectionMeta = await new Promise<CollectionMeta>((resolve, reject) => {
+            client.createCollection(this.link, collectionDef, options, (err, result) => {
+                err ? reject(err) : resolve(result);
+            });
+        });
 
-        throw new UserCancelledError();
+        return this.initChild(collection);
     }
 
     private static validatePartitionKey(key: string): string | undefined | null {
-        if (/^[#?\\]*$/.test(key)) {
+        if (/[#?\\]/.test(key)) {
             return "Cannot contain these characters - ?,#,\\, etc.";
```

Comment:
Please change " - " to colon:

```suggestion
            return "Cannot contain these characters: ?,#,\\, etc.";
```